src/helpers.rs

```diff
@@ -304,4 +304,29 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     fn eval_libc_i32(&mut self, name: &str) -> InterpResult<'tcx, i32> {
         self.eval_libc(name)?.to_i32()
     }
+
+    /// Sets the last error variable
+    fn set_last_error(&mut self, scalar: Scalar<Tag>) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+        let errno_place = this.machine.last_error.unwrap();
+        this.write_scalar(scalar, errno_place.into())
+    }
+
+    /// Gets the last error variable
+    fn get_last_error(&mut self) -> InterpResult<'tcx, Scalar<Tag>> {
+        let this = self.eval_context_mut();
+        let errno_place = this.machine.last_error.unwrap();
+        this.read_scalar(errno_place.into())?.not_undef()
+    }
+
+    /// Sets the last error variable using a `std::io::Error`. It fails if the error cannot be
+    /// transformed to a raw os error succesfully
+    fn set_last_error_from_io_error(&mut self, e: std::io::Error) -> InterpResult<'tcx> {
+        self.eval_context_mut().set_last_error(Scalar::from_int(
+            e.raw_os_error().ok_or_else(|| {
```

Comment:
One thing I realized: when calling `raw_os_error`, you are relying on the host and target OS being the same. When running this on Windows with a Linux target, the result will be totally wrong.

For now I am fine not supporting this case, but I feel we should error as "unsupported" instead of silently returning bad error codes.

Reply:
Hmmm... that's true. I could change the method to match the error and return the proper error code.

Reply:
@RalfJung should i do that? ^

Reply:
Yeah that's probably the most robust thing to do.

Reply:
I think that we might have a problem with that here. Not all `io::Error`s are generated from an OS error code. So actually it might be a better idea to try to translate the error codes themselves.

Reply:
I have no idea what you are talking about... "math the error" = "translate the error code", no?

To determine the target platform, see

https://github.com/rust-lang/miri/blob/089c7e82596abdbc0fd5f21fee9e9438b453ec65/src/eval.rs#L217

Reply:
I'll try to fix this for unix first

Reply:
Check: https://github.com/rust-lang/miri/pull/991/commits/70e76489b1c9708db596de885b9fbdfd96a5af29. I was thinking in doing crazy stuff between the platforms' OS errors but I think just taking the `ErrorKind` is good enough.